docs/design/configs.md

```diff
@@ -0,0 +1,130 @@
+# Generic Config Object
```

Comment:
`Paramerter` or `Argument` makes it seem like it's only related to command-line flags. @bgrant0607 is also changed his mind about `Parameter` I believe.